speech/api/Recognize/InfiniteStreaming.cs

```diff
@@ -0,0 +1,248 @@
+        /// </summary>
+        private SpeechClient.StreamingRecognizeStream _rpcStream;
+
+        /// <summary>
+        /// The deadline for when we should stop the current stream.
+        /// </summary>
+        private DateTime _rpcStreamDeadline;
+
+        /// <summary>
+        /// The task indicating when the next response is ready.
+        /// </summary>
+        private Task<bool> _moveNextTask;
+
+        private InfiniteStreaming()
+        {
+            _client = SpeechClient.Create();
+        }
+
+        /// <summary>
+        /// Runs the main loop until "exit" or "quit" is heard.
+        /// </summary>
+        private async Task RunAsync()
+        {
+            using (var microphone = StartListening())
+            {
+                while (true)
+                {
+                    await MaybeStartStreamAsync();
+                    // ProcessResponses will return false if it hears "exit" or "quit".
+                    if (!ProcessResponses())
+                    {
+                        return;
+                    }
+                    await TransferMicrophoneChunkAsync();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Starts a new RPC streaming call if necessary. This will be if either it's the first call
+        /// (so we don't have a current request) or if the current request will time out soon.
+        /// In the latter case, after starting the new request, we copy any chunks we'd already sent
+        /// in the previous request which hadn't been included in a "final result".
+        /// </summary>
+        private async Task MaybeStartStreamAsync()
+        {
+            var now = DateTime.UtcNow;
+            if (_rpcStream != null && now >= _rpcStreamDeadline)
+            {
+                Console.WriteLine($"Closing stream before it times out");
+                await _rpcStream.WriteCompleteAsync();
+                _rpcStream.GrpcCall.Dispose();
+                _rpcStream = null;
+            }
+
+            // If we have a valid stream at this point, we're fine.
+            if (_rpcStream != null)
+            {
+                return;
+            }
+            // We need to create a new stream, either because we're just starting or because we've just closed the previous one.
+            _rpcStream = _client.StreamingRecognize();
+            _rpcStreamDeadline = now + s_streamTimeLimit;
+            _processingBufferStart = TimeSpan.Zero;
+            _moveNextTask = _rpcStream.ResponseStream.MoveNext();
+            await _rpcStream.WriteAsync(new StreamingRecognizeRequest
+            {
+                StreamingConfig = new StreamingRecognitionConfig
+                {
+                    Config = new RecognitionConfig
+                    {
+                        Encoding = RecognitionConfig.Types.AudioEncoding.Linear16,
+                        SampleRateHertz = SampleRate,
+                        LanguageCode = "en-US",
```

Comment:
Could we take language code from user input [ as parameter ]? my suggestion

Reply:
While that would be entirely reasonable, I think I'd prefer to do that for all the Speech samples rather than make this one different from everything else. I'd like to get this one in first, then we can consider updating everything to allow a language code to be specified.

Reply:
Okay sounds good to me